### PR TITLE
Adding content about out of stock products in recs

### PR DESCRIPTION
**Pull request by erikmarr**
## Purpose of this pull request

This pull request (PR) adds information about the behavior of recommendations with the **Display Out-of-Stock Products** option in the Catalog configuration.

If Magento Commerce, please specify:

- [x] Commerce only?
- [ ] Commerce with B2B?

## Affected documentation pages

- https://docs.magento.com/user-guide/marketing/product-recommendations.html

### CODE REVIEW

File: src/marketing/product-recommendations.md [hunk 1]
```diff
@@ -65,6 +65,10 @@ The following table describes how the **Most popular**, **Trending**, and **Reco
 |Cart|Categories of the products in the shopper's cart|
 |Order Confirmation|Categories for products the shopper just purchased|
 
+### Filter based on stock
```
Comment (misha-kotov) [inline src/marketing/product-recommendations.md:68]:
```suggestion
### Filter based on stock status
```

File: src/marketing/product-recommendations.md [hunk 2]
```diff
@@ -65,6 +65,10 @@ The following table describes how the **Most popular**, **Trending**, and **Reco
...
+
+Product Recommendations use the **Display Out-of-Stock Products** value configured in the Admin UI to determine if a particular product should be displayed in a recommendation unit. If the **Display Out-of-Stock Products** option is set to **Yes**, products that are out-of-stock will be displayed in the recommendation unit. If **Display Out-of-Stock Products** option is set to **No**, out-of-stock products will not be displayed in recommendation units. You can configure the **Display Out-of-Stock Products** value in the [stock options]({% link configuration/catalog/inventory.md %}) section of the Admin UI.
```
Comment (misha-kotov) [inline src/marketing/product-recommendations.md:70]:
```suggestion
Product Recommendations use the **Display Out-of-Stock Products** value configured in the Magento Admin Panel to determine if a particular product should be displayed in a recommendation unit. If the **Display Out-of-Stock Products** option is set to **Yes**, products that are out-of-stock are eligible to be displayed in the recommendation unit. If **Display Out-of-Stock Products** option is set to **No**, out-of-stock products will not be displayed in recommendation units. You can configure the **Display Out-of-Stock Products** value in the [stock options]({% link configuration/catalog/inventory.md %}) section of the Configuration.
```

File: src/marketing/product-recommendations.md [hunk 3]
```diff
@@ -65,6 +65,10 @@ The following table describes how the **Most popular**, **Trending**, and **Reco
 |Cart|Categories of the products in the shopper's cart|
 |Order Confirmation|Categories for products the shopper just purchased|
 
+### Filter based on stock status
+
+Product Recommendations use the **Display Out-of-Stock Products** value configured in the Magento Admin Panel to determine if a particular product should be displayed in a recommendation unit. If the **Display Out-of-Stock Products** option is set to **Yes**, products that are out-of-stock are eligible to be displayed in the recommendation unit. If **Display Out-of-Stock Products** option is set to **No**, out-of-stock products will not be displayed in recommendation units. You can configure the **Display Out-of-Stock Products** value in the [stock options]({% link configuration/catalog/inventory.md %}) section of the Configuration.
```
Comment (leslietilling) [inline src/marketing/product-recommendations.md:70]: Still looks kind of odd with Display Out-of-Stock bolded three times in the same paragraph. 

```suggestion
Product Recommendations use the **Display Out-of-Stock Products** value configured in the Magento Admin to determine if a particular product should be displayed in a recommendation unit. If the **Display Out-of-Stock Products** option is set to `Yes`, products that are out-of-stock are eligible to be displayed in the recommendation unit. If **Display Out-of-Stock Products** option is set to `No`, out-of-stock products will not be displayed in recommendation units. You can configure the **Display Out-of-Stock Products** value in the [stock options]({% link configuration/catalog/inventory.md %}) section of the Catalog configuration.
```

**Comment by erikmarr** [comment]
running tests

**Pull request merged**